src/components/NewMemberSection/index.tsx

```diff
@@ -1,21 +1,48 @@
+import { useEffect, useRef } from 'react';
 import { useGetUserWithLoadMore } from '@/src/services/serverApi';
 import NewMemberSectionPresentation from './Presentation';
 import { Button } from '@chakra-ui/react';
-import styles from './newMemberSection.module.css'
+import styles from './newMemberSection.module.css';
 
 export default function NewMemberSection() {
-  const { data, isLoading, isFetching, loadMore, links } = useGetUserWithLoadMore()
+  const { data, isLoading, isFetching, loadMore, links } =
+    useGetUserWithLoadMore();
+  const loadMoreRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (!loadMoreRef.current) return;
+
+    const observer = new IntersectionObserver(
+      (entries) => {
+        if (entries[0].isIntersecting && links?.next) {
+          loadMore();
+        }
+      },
+      {
+        root: null,
+        rootMargin: '0px',
+        threshold: 1.0,
+      },
+    );
+
+    observer.observe(loadMoreRef.current);
+
+    return () => {
+      if (loadMoreRef.current) {
+        observer.unobserve(loadMoreRef.current);
+      }
+    };
+  }, [loadMore, links]);
 
   return (
     <div>
-      <NewMemberSectionPresentation
-        data={data}
-        isLoading={isLoading}
-      />
-
-      {links?.next && links.next.length &&
-        <div className={styles.newMemberSectionLoadMore__container}>
+      <NewMemberSectionPresentation data={data} isLoading={isLoading} />
 
+      {links?.next && links.next.length && (
+        <div
+          className={styles.newMemberSectionLoadMore__container}
+          ref={loadMoreRef}
+        >
```

Comment:
Since it is going to be infinitely scrolled, do we still need to keep the button on line 46?

Reply:
This button provides a visual indication to users that new data is being loaded, which can be especially helpful if their internet connection is slow. It improves the user experience by giving them feedback that the system is working on loading more content, rather than leaving them wondering if the scroll is stuck or if the content is simply not available. This kind of feedback can make the interface feel more responsive and user-friendly. Please correct me if I'm wrong here.

Reply:
I tried doing some research and wrote a doc as well. So, most of the platforms show a spinner instead of button like component. Do you think I should replace this with a spinner? 
For ref: https://docs.google.com/document/d/1Zp5do4tx3sNKBc9IZJ8JDYNbj4aJ6tBLvrDxi_GUZu8/edit

Reply:
Yes, in my opinion, spinner would make more sense here. Is there an existing spinner component being used at any other place that can be re-used? Also, would like to know @iamitprakash 's thoughts for this?

Reply:
I checked our existing packages and found that Chakra UI provides a spinner component, so I used it for the spinner loader. You can see the preview.  Please let me know if any changes are needed. Thanks for reviewing my PR!
<img width="1680" alt="Screenshot 2024-08-22 at 22 39 34" src="https://github.com/user-attachments/assets/b1a3210c-8791-4159-996c-c106514dd153">
@hjaintech